tcpproxy.go

```diff
@@ -108,8 +132,25 @@ func (p *TCPProxy) isClosed() bool {
 	return p.closed
 }
 
-func stream(dst io.WriteCloser, src io.ReadCloser) {
+func (p *TCPProxy) stream(dst io.WriteCloser, src io.ReadCloser) {
 	defer dst.Close()
 	defer src.Close()
-	io.Copy(dst, src)
+	buf := make([]byte, 32*1024)
+	for {
+		nr, er := src.Read(buf)
```

Comment:
Since you're not bothered about the actual errors, I'd name the error variables just as conventional "err".